### DISCUSSION TITLE
how to set python dll path?

### DISCUSSION THREAD

**Discussion by mhyeonsoo** [pythonnet/pythonnet#1684] [Q&A]
### Environment

-   Pythonnet version: 3.0.0
-   Python version: 3.7
-   Operating System: Mac OS 10.15.7
-   .NET Runtime:

### Details

-   Describe what you were trying to get done.

    I followed  https://github.com/pythonnet/pythonnet#embedding-python-in-net. to setup pythonnet with unity project,
   and also setup the dylib path in the cs script.

    I am building unity project and trying to run it on Android device.
    Although I have linked the dll file, it seems like the project cannot detect it. 
    When I run the compiled app, I could see the error message through adb logcat.

    ```
    DllNotFoundException: Could not load /-/opt/anaconda3/lib/libpython3.9.dylib with flags RTLD_NOW | 
    RTLD_GLOBAL: dlopen failed: invalid flags to dlopen
    ```
    could you help me with setting Python.Runtime.dll file appropriately?
    Thanks,

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

```csharp

    string pythonPath1 = @"/-/opt/anaconda3/bin/python3.9";
    string pythonPath2 = @"/-/opt/anaconda3/lib/python3.9/site-packages";
    Environment.SetEnvironmentVariable("PATH", pythonPath1, EnvironmentVariableTarget.Process);
    Environment.SetEnvironmentVariable("PYTHONHOME", pythonPath1, EnvironmentVariableTarget.Process);
    Environment.SetEnvironmentVariable("PYTHONPATH", pythonPath2, EnvironmentVariableTarget.Process);
    Environment.SetEnvironmentVariable("PYTHONNET_PYDLL", @"/-/opt/anaconda3/lib/libpython3.9.dylib");    

    using (Py.GIL())
        {
            dynamic np = Py.Import("numpy");
            Debug.Log(np.cos(np.pi * 2));
            

            dynamic sin = np.sin;
            Debug.Log(sin(5));

            double c = np.cos(5) + sin(5);
            Debug.Log(c);

            dynamic a = np.array(new List<float> { 1, 2, 3 });
            Debug.Log(a.dtype);

            dynamic b = np.array(new List<float> { 6, 5, 4 }, dtype: np.int32);
            Debug.Log(b.dtype);

            Debug.Log(a * b);
        }
```

-   If there was a crash, please include the traceback here.

```python
    print('TODO')
```

**Comment by CodingMadness** [comment]
@mhyeonsoo 

here is how it works: 

`
string pyPath = @"/-/opt/anaconda3/bin/python3.9";
Runtime.PythonDLL = pyPath;
`

**Reply by barnabaspal** [reply]
How to set Runtime.PythonDLL? I got "CS0198	A static readonly field cannot be assigned to (except in a static constructor or a variable initializer)" build error in my project (net core app).

**Reply by lostmsu** [reply]
@barnabaspal you are using 2.5. `PythonDLL` property was first exposed in 3.0 betas.

**Comment by mhyeonsoo** [comment]
@Shpendicus 

thanks for your comment.
I will check that out!
